src/components/Menu.tsx

```diff
@@ -555,11 +555,36 @@ export function Menu() {
         sorted.sort((a, b) => moment(b.modified).diff(moment(a.modified)));
         break;
     }
+  }
 
-    const folders = sorted.filter((item) => item.type === "folder");
-    const files = sorted.filter((item) => item.type !== "folder");
+  const folders = sorted.filter((item) => item.type === "folder");
+  const files = sorted.filter((item) => item.type !== "folder");
 
-    return [...folders, ...files];
+  return [...folders, ...files];
+};
+
+export function Menu() {
+  const api = new ControlApi();
+
+  const [currentPath, setCurrentPath] = useState<string>("");
+  const [sort, setSort] = useState<Sort>(Sort.NameAsc);
+
+  const listMenuFolder = useListMenuFolder(currentPath);
+  const [sortedItems, setSortedItems] = useState<MEMenuItem[]>(listMenuFolder.data?.items || []);
+
+  useLayoutEffect(() => {
+    setSortedItems(sortItems(listMenuFolder.data?.items, sort, currentPath));
+  }, [sort, currentPath, listMenuFolder.data?.items]);
```

Comment:
sort items only when those 3 parameters change, all the other time just render the stored array